### PR TITLE
upgrade commonmark version

### PR DESCRIPTION
**Pull request by harikt**
Tested on php 8.1 on local.

### CODE REVIEW

**Comment by harikt** [comment]
This is not complete. I am still experimenting with this.

**Comment by pmjones** [comment]
@harikt Thanks for this (and for all the other work you've done lately).

When you are ready, this should target the 2.x branch, which I will bring up-to-date.

I think 2.x ought to target PHP 8.1 and up -- any thoughts on that?

Thanks again!

**Comment by harikt** [comment]
8.1 for 2.x seems good to me. 

Regardin this PR, it may take sometime than expected.

**Comment by pmjones** [comment]
Understood. I will try working on it as well.

FWIW, the webuni commonmark extensions have (I think) been consolidated into the CommonMark 2.

**Comment by harikt** [comment]
You are correct. It is in 1.5 x onwards.

**Comment by harikt** [comment]
Two failures 

1. https://gist.github.com/harikt/faba7ec0039b8a016f251fcf82ed1538/revisions

2. 
```
--- Expected
+++ Actual
@@ @@
 <p>Text under sub-subtitle.</p>
 <h2>Subtitle B</h2>
 <p>Text under subtitle B.</p>
-<blockquote title="Blockquote title">
-<p>Blockqoute</p>
+<blockquote>
+<p>Blockqoute
+{: title="Blockquote title"}</p>
 </blockquote>
-<table>
-<thead>
-<tr>
-<th>th</th>
-<th align="center">th(center)</th>
-<th align="right">th(right)</th>
-</tr>
-</thead>
-<tbody>
-<tr>
-<td>td</td>
-<td align="center">td</td>
-<td align="right">td</td>
-</tr>
-</tbody>
-</table>
+<p>th | th(center) | th(right)
+---|:----------:|----------:
+td |     td     |         td</p>
```

I am not sure if we can change the expected content to pass or not. I will do some research on this and if I can I will fix else if you can please do.

Thank you.

File: tests/BookFixture.php [hunk 1]
```diff
@@ -17,12 +17,6 @@ class BookFixture
         "templates": {
             "foo": "/foo.php"
         },
-        "extensions": {
```
Comment (harikt) [inline tests/BookFixture.php:20]: Probably we can bring this back. Currently I have added these extensions by default.

File: src/Process/Conversion/ConversionProcessBuilder.php [hunk 2]
```diff
@@ -73,9 +74,11 @@ protected function newCommonMarkConverter(RootConfig $config)
...
 
-        return new Converter(
-            new DocParser($environment),
-            new HtmlRenderer($environment)
-        );
+        // Finally add common mark extension
+        $environment->addExtension(new CommonMarkCoreExtension());
+        $environment->addExtension(new AttributesExtension());
+        $environment->addExtension(new TableExtension());
+
```
Comment (harikt) [inline src/Process/Conversion/ConversionProcessBuilder.php:81]: We can probably remove all these extensions. In that case we can switch between commonmark and GitHub flavoured markdown.

**Comment by harikt** [comment]
I guess it is safe to merge to 1.x itself. There is no breaking tests.

**Comment by pmjones** [comment]
> I guess it is safe to merge to 1.x itself.

The problem is, when you change a dependency to another major version, that itself is a BC-breaking change -- so technically it ought to go to 2.x. :-/

**Comment by pmjones** [comment]
@harikt I think this is good! And it keeps the PHP version all the way back to 5.6, which is also nice.

Do you care if I merge this PR to 2.x, instead of 1.x? 2.x can stay at PHP 5.6, and maybe a 3.x version can bring it up to PHP 8.1. (The number of changes to use the 8.1 feature set would take quite a while, I think, and I know we want to release this to support the recent Aura releases.)

**Comment by harikt** [comment]
I agree with your thoughts on BC break. In that case there is one more change we probably want to do. Just thought of that. Changing the require of league/common mark to support 1.x || 2.x . Else those major versions may be restricting the php version constraint.

Another thought I was having for common mark is technically changing 1
X to 2.x is a BC break. But common mark is following somthing similar to RFC specification. So if that spec is the same there is no issue probably.

**Comment by pmjones** [comment]
@harikt Right on.

I will merge this PR as-is to the 2.x branch, then make some further modifications and test it on prior PHP versions, back to 7.1 if I can (since that's what CommonMark 1.x requires).

Thanks for the good work here!

**Comment by pmjones** [comment]
@harikt Local experimenting shows me that CommonMark 1.x and 2.x do have strong BC breaks between them, and the conversion/rending processors have to be targeted to one or the other. Unless you feel strongly about it, I plan to target 2.x. In turn, that means the PHP versions will have to be `^7.4||^8.0`.

**Comment by harikt** [comment]
I have no objection. I am already running on 8.x and 7.4

**Comment by pmjones** [comment]
@harikt All right -- I think 2.x is ready for a 2.0.0 release, with all your changes and a few more. If you would, please try building a project using it (composer require would be 2.x-dev) and let me know if you see any problems. If everything goes well, I'll roll the 2.0.0 release. Thanks for your help, and for your patience!

**Comment by harikt** [comment]
The project we can try out is aura itself. I will try when I get some time tomorrow.

**Comment by harikt** [comment]
@pmjones aura docs are generated via 2.x https://github.com/auraphp/auraphp.github.com/blob/54a6e48f2d90f68020b63730cffdbf96221ffe79/_bookdown/composer.json#L21 . No issues noticed so far.

**Pull request merged**